jadx-gui/src/main/java/jadx/gui/ui/QuarkDialog.java

```diff
@@ -0,0 +1,232 @@
+		}
+		return arr;
+	}
+
+	public final void initUI() {
+
+		JLabel description = new JLabel("Analyzing apk using Quark-Engine");
+		JLabel selectApkText = new JLabel("Select Apk/Dex");
+		description.setAlignmentX(0.5f);
+
+		selectFile = new JComboBox<String>(filesToStringArr());
+
+		JPanel textPane = new JPanel();
+
+		textPane.add(description);
+
+		JPanel selectApkPanel = new JPanel();
+		selectApkPanel.add(selectApkText);
+		selectApkPanel.add(selectFile);
+
+		progressPane = new JPanel();
+		progressPane.setVisible(false);
+		progressPane.setSize(150, 10);
+
+		progressBar = new JProgressBar(0, 100);
+		progressBar.setSize(150, 10);
+		progressBar.setIndeterminate(true);
+		progressBar.setStringPainted(false);
+		progressPane.add(progressBar);
+
+		JPanel buttonPane = new JPanel();
+		JButton start = new JButton("Start");
+		JButton close = new JButton(NLS.str("tabs.close"));
+		close.addActionListener(event -> close());
+		start.addActionListener(event -> analyzeAPK());
+		buttonPane.add(start);
+		buttonPane.add(close);
+		getRootPane().setDefaultButton(close);
+
+		JPanel centerPane = new JPanel();
+		centerPane.add(selectApkPanel);
+		centerPane.add(progressPane);
+		Container contentPane = getContentPane();
+
+		contentPane.add(textPane, BorderLayout.PAGE_START);
+		contentPane.add(centerPane);
+		contentPane.add(buttonPane, BorderLayout.PAGE_END);
+
+		setTitle("Quark Engine");
+		pack();
+		setSize(200, 125);
+		setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+		setModalityType(ModalityType.MODELESS);
+		setLocationRelativeTo(null);
+	}
+
+	private void analyzeAPK() {
+		LoadTask task = new LoadTask();
+		task.execute();
+	}
+
+	private void loadReportFile() {
+		try {
+			JsonObject quarkReport = (JsonObject) JsonParser.parseReader(new FileReader(quarkReportFile.getAbsolutePath().toString()));
+
+			JRoot root = mainWindow.getCacheObject().getJRoot();
+
+			QuarkReport quarkNode = QuarkReport.analysisAPK(quarkReport);
+
+			root.update();
+			root.add(quarkNode);
+
+			mainWindow.reloadTree();
+
+		} catch (JsonIOException | JsonSyntaxException | FileNotFoundException e) {
+			LOG.error("Quark: Load report failed: ", e);
+		}
+
+	}
+
+	private void close() {
+		dispose();
+	}
+
+	@Override
+	public void dispose() {
+		LogCollector.getInstance().resetListener();
+		settings.saveWindowPos(this);
+		super.dispose();
+	}
+
+	private class LoadTask extends SwingWorker<Void, Void> {
+		public LoadTask() {
+			progressPane.setVisible(true);
+		}
+
+		@Override
+		public Void doInBackground() {
+			try {
+
+				quarkReportFile = File.createTempFile("QuarkReport-", ".json");
+
+				String outputPath = quarkReportFile.getAbsolutePath().toString();
+				String apkName = selectFile.getSelectedItem().toString();
+				String apkPath = null;
+				for (Path path : files) {
+					if (path.getFileName().toString().equals(apkName)) {
+						apkPath = path.toString();
+					}
+				}
+				String cmd = "quark -a " + apkPath + " -s -o " + outputPath;
```

Comment:
@pulorsok 
Potential command injection. Please use String array to prevent this

Reply:
@bagipro thanks for check, but unfortunately I already merged this PR :disappointed: 
I will fix these myself, do you have any links how to do it correctly? Only replacing with array is enough?

Reply:
@skylot 
Yeah, the array is enough

Reply:
@bagipro thanks! Will fix ASAP

Reply:
Fixed in 52412dfe31a1184a512636349cbe3a68792ecb88